packages/ai/src/authorizers/federated-connections/FederatedConnectionAuthorizerBase.ts

```diff
@@ -128,24 +145,59 @@ export class FederatedConnectionAuthorizerBase<ToolExecuteArgs extends any[]> {
 
     const { connection } = store;
 
-    const subjectToken = await this.getRefreshToken(...toolContext);
+    // Determine which token type to use and get the appropriate subject token
+    let subjectToken: string | undefined;
+    let subjectTokenType: string;
+    const clientId: string =
+      this.auth0.clientId || this.auth0.resourceServerClientId!;
+    const clientSecret: string =
+      this.auth0.clientSecret || this.auth0.resourceServerClientSecret!;
+
+    if (typeof this.params.refreshToken === "function") {
+      subjectToken = await this.getRefreshToken(...toolContext);
+      subjectTokenType = "urn:ietf:params:oauth:token-type:refresh_token";
+    } else if (typeof this.params.accessToken === "function") {
+      subjectToken = await resolveParameter(
+        this.params.accessToken,
+        toolContext
+      );
+      subjectTokenType = "urn:ietf:params:oauth:token-type:access_token";
+    } else {
+      // This should never happen due to constructor validation, but TypeScript needs this
+      throw new Error("Either refreshToken or accessToken must be configured");
+    }
 
     if (!subjectToken) {
       return;
     }
 
-    const exchangeParams = {
+    // Get optional login hint if provided
+    const loginHint =
+      typeof this.params.loginHint === "function"
+        ? await resolveParameter(this.params.loginHint, toolContext)
+        : undefined;
+
+    const exchangeParams: Record<string, string> = {
       grant_type:
         "urn:auth0:params:oauth:grant-type:token-exchange:federated-connection-access-token",
-      client_id: this.auth0.clientId,
-      client_secret: this.auth0.clientSecret,
-      subject_token_type: "urn:ietf:params:oauth:token-type:refresh_token",
+      client_id: clientId,
+      subject_token_type: subjectTokenType,
       subject_token: subjectToken,
       connection: connection,
       requested_token_type:
         "http://auth0.com/oauth/token-type/federated-connection-access-token",
     };
 
+    // Add client secret if available
+    if (clientSecret) {
+      exchangeParams.client_secret = clientSecret;
+    }
+
+    // Add login hint if provided
+    if (loginHint) {
+      exchangeParams.login_hint = loginHint;
+    }
```

Comment:
passing the `user.sub` as the login_hint seems to currently return the error thrown here in server:
https://github.com/atko-cic/auth0-server/blob/c655ccc60b4dd583a5ec35d31885838a5b2a046b/packages/server/lib/federated_connection_at/token_exchange.js#L468-L471
```
{
  error: "user_not_found",
  error_description: "Identity User not found.",
}
```